newroutes/__tests__/ContactUs.spec.js

```diff
@@ -0,0 +1,121 @@
+const express = require("express")
+const _ = require("lodash")
+const request = require("supertest")
+
+const { errorHandler } = require("@middleware/errorHandler")
+const { attachReadRouteHandlerWrapper } = require("@middleware/routeHandler")
+
+const { contactUsContent, contactUsSha } = require("@fixtures/contactUs")
+const { generateRouter } = require("@root/fixtures/app")
+
+const { ContactUsRouter } = require("../contactUs")
+
+describe("ContactUs Router", () => {
+  const mockContactUsPageService = {
+    read: jest.fn(),
+    update: jest.fn(),
+  }
+
+  const router = new ContactUsRouter({
+    contactUsPageService: mockContactUsPageService,
+  })
+
+  const subrouter = express()
+  // We can use read route handler here because we don't need to lock the repo
+  subrouter.get(
+    "/:siteName/contactUs",
+    attachReadRouteHandlerWrapper(router.readContactUs)
+  )
+  subrouter.post(
+    "/:siteName/contactUs",
+    attachReadRouteHandlerWrapper(router.updateContactUs)
+  )
+  const app = generateRouter(subrouter)
+
+  const siteName = "test-site"
+  const accessToken = undefined // Can't set request fields - will always be undefined
+
+  const reqDetails = { siteName, accessToken }
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe("readContactUs", () => {
+    const expectedResponse = {
+      sha: contactUsSha,
+      content: contactUsContent,
+    }
+    mockContactUsPageService.read.mockResolvedValue(expectedResponse)
+
+    it("retrieves contactUs details", async () => {
+      const resp = await request(app).get(`/${siteName}/contactUs`).expect(200)
+
+      expect(resp.body).toStrictEqual(expectedResponse)
+      expect(mockContactUsPageService.read).toHaveBeenCalledWith(reqDetails)
+    })
+  })
+
+  describe("updateContactUs", () => {
+    it("rejects requests with invalid body", async () => {
+      await request(app).post(`/${siteName}/contactUs`).send({}).expect(400)
+    })
+
+    it("rejects requests with incomplete body", async () => {
+      // Missing layout
+      const incompleteDetails = _.cloneDeep(contactUsContent)
+      const reqBody = {
+        content: incompleteDetails,
+        sha: contactUsSha,
+      }
+      delete incompleteDetails.frontMatter.layout
```

Comment:
wdyt about using [`_.omit`](https://lodash.com/docs/4.17.15#omit) here?

Reply:
Great suggestion, I missed out that functionality - https://github.com/isomerpages/isomercms-backend/pull/349/commits/1d049826623b36a4708e27b6056792652beb24fa